### PR TITLE
feat: add notes

### PR DESCRIPTION
**Pull request by Chiragmodi01**
[Netlify Status](https://app.netlify.com/sites/tranquil-choux-2fb728/deploys)

### Main Features:
- User can add a new note.
- User can clear note's textarea
- User can pin/unpin notes.
- User can search for any note.

### Other Features:
- User can change app's theme.
- User can close side panel.

_there will be few minor fixes in the ui scheme!_

notero

(_the first 6 files can be skipped_)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,30 @@
 import "./App.css";
+import React, { useState } from 'react'; 
+import Mockman from "mockman-js";
 import Homepage from "./pages/Homepage";
+import Labels from "./pages/Labels";
+import Archive from "./pages/Archive";
+import Trash from "./pages/Trash";
```
Comment (ravikumar0403) [inline src/App.js]: You can use [barrel imports](https://dev.to/justmyrealname/organize-react-components-better-with-barrel-exports-4fno) here to make your code cleaner

File: src/assets/assets.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 import notero from './notero.svg';
+import noteroLight from './noteroLight.svg';
```
Comment (ravikumar0403) [inline src/assets/assets.js]: this can be written as `export default as _somename_ from "path/to/file"`

File: src/comps/CreateNotes.jsx [hunk 3]
```diff
@@ -1,12 +1,103 @@
...
 import '../styles/homepage/create_notes.css'
+import { MdOutlineArchive, MdOutlineLabel } from 'react-icons/md';
+import { IoColorPaletteOutline } from 'react-icons/io5';
+import { AiOutlineClear } from 'react-icons/ai'
+import { BsPin, BsPinFill } from 'react-icons/bs'
+import TextareaAutosize from 'react-textarea-autosize';
+import { useNote } from '../helpers/context/note-context';
+import { postNoteService } from '../helpers/services/postNoteService';
+
+function CreateNotes({ createNote, setCreateNote, createNoteRef }) {
```
Comment (ravikumar0403) [inline src/comps/CreateNotes.jsx:11]: i would suggest using arrow functions for creating components

File: src/comps/CreateNotes.jsx [hunk 4]
```diff
@@ -1,12 +1,103 @@
...
+    if(note.desc === ""){
+      toast.error("Note's Content cannot be empty!", {
+        position: "bottom-left",
+        autoClose: 2000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+        });
```
Comment (ravikumar0403) [inline src/comps/CreateNotes.jsx]: If these options are common for all toast then you can pass it as a prop to the `<ToastContainer />` and skip here

File: src/comps/Navbar.jsx [hunk 5]
```diff
@@ -1,26 +1,72 @@
...
-            <input type="text" className="search-input" placeholder='Search'/>
-            <BsFilter size="2em" className='filter-icon'/>
+            <FiSearch size='1.4em' className={searchFocus ? 'search-icon dark': 'search-icon'}/>
+            <input value={search} type="text" className={searchFocus ? 'search-input dark': 'search-input'} placeholder='Search' ref={searchFocusRef} onClick={(e) => focusSearchHandler(e)} onChange={(e) => setSearch(e.target.value)}/>
+            <BsFilter size="2em" className={searchFocus ? 'filter-icon dark': 'filter-icon'}/>
         </div>
+        {utilsState.darkTheme && <BsSun title="Light theme" size='1.7em' className='ham-menu_icon theme' onClick={() =>themeClickHandler()}/>}
+        {!utilsState.darkTheme && <BsMoon title="Dark theme" size='1.7em' className='ham-menu_icon theme' onClick={() => themeClickHandler()}/>}
         </div>
+        <ToastContainer
```
Comment (ravikumar0403) [inline src/comps/Navbar.jsx]: you can have one common `<ToastContainer />` in App.js, you don't need it in every component

File: src/comps/NoteCard.jsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    })
+    notesDispatch({type: 'UPDATE_NOTE', payload: payload})
+  }
+
+  return (
+    <div className='NoteCard' key={id}>
+        <div className="noteCard_head">
+          <h3 className='noteCard_head_title'>{title}</h3>
+          { pin && <BsPinFill size="1.4em" className='noteCard_icon' title="Unpin note" onClick={() => notePinToggler()}/>}
+          { !pin && <BsPin size="1.4em" className='noteCard_icon' title="Pin note" onClick={() => notePinToggler()}/>}
```
Comment (ravikumar0403) [inline src/comps/NoteCard.jsx]: this block can be written with ternary operator

File: src/comps/Sidenav.jsx [hunk 7]
```diff
@@ -1,33 +1,37 @@
...
 import { MdOutlineArchive, MdOutlineDelete, MdOutlineLabel } from 'react-icons/md';
 import { BiUser } from 'react-icons/bi';
 import { IoExitOutline } from 'react-icons/io5';
+import { useNote } from '../helpers/context/note-context';
 
-function Sidenav({ hideAside }) {
+function Sidenav( ) {
+  const { utilsState } = useNote()
   
   const location = useLocation();
```
Comment (ravikumar0403) [inline src/comps/Sidenav.jsx]: you could have destructured pathname here itself.. `const { pathname } = useLocation()`

File: src/comps/Sidenav.jsx [hunk 8]
```diff
@@ -1,33 +1,37 @@
...
   const currPath = location.pathname;
+  let navigate = useNavigate();
 
   return (
     <div className='Sidenav'>
-      <div value={hideAside} className={hideAside ? "title_hidden sidenav_container" : "sidenav_container"}>
+      <div className={utilsState.hideAside ? "title_hidden sidenav_container" : "sidenav_container"}>
         <div className="sidenav_top">
-          <div className={currPath === '/Home' ? 'active_menu sidenav_menu' : 'sidenav_menu'}>
+          <div className={currPath === '/' ? 'active_menu sidenav_menu' : 'sidenav_menu'} onClick={() => navigate("/")}>
```
Comment (ravikumar0403) [inline src/comps/Sidenav.jsx]: Do checkout [NavLink](https://reactrouter.com/docs/en/v6/api#navlink) component from `react-router-dom`, `By default, an active class is added to a <NavLink> component when it is active.` which you can use for styling

File: src/helpers/context/note-context.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        hideAside: false,
+        pinNote: false,
+        darkTheme: true
+    })
+
+    const [notesState, notesDispatch] = useReducer(notesReducer, {
+        notes: [],
+        filteredNotes: [],
+        // pinnedNotes: [],
+        // unpinnedNotes: [],
```
Comment (ravikumar0403) [inline src/helpers/context/note-context.js]: remove commented code if not required

File: src/helpers/services/postNoteService.js [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+const postNoteService = async(notesDispatch, note, setNote) => {
+    const encodedToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI4ZTc5YTI1Ni01ZmQ2LTQwZTktYmYwYS04NGUyNGE5MDNmYzAiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.tPP0Mb50-2ihtbeWd7PUpf1_pkzVQjmcxNVBIuqzFlU"
+
+    const Headers = { authorization: encodedToken}
+
+    const noteToSend = {note : note}
+    try {
+        const res = await axios.post("/api/notes", noteToSend, {headers: Headers});
+        console.log(res.status, 'post')
+        console.log(res.data.notes, 'post')
```
Comment (ravikumar0403) [inline src/helpers/services/postNoteService.js:12]: Always remove console logs before pushing the code

**Review by ravikumar0403 — APPROVED** [review]

File: src/comps/Navbar.jsx (hunk 5 above)
Comment (Chiragmodi01) [reply src/comps/Navbar.jsx]: Yes, was planning create reusable comp for toast.

File: src/helpers/services/postNoteService.js (hunk 10 above)
Comment (Chiragmodi01) [reply src/helpers/services/postNoteService.js:12]: cool, thanks for all inputs.

File: src/App.js (hunk 1 above)
Comment (Chiragmodi01) [reply src/App.js]: will look into that

File: src/comps/CreateNotes.jsx (hunk 3 above)
Comment (aurobindoGupta) [inline src/comps/CreateNotes.jsx:11]: start using es6 preferably

File: src/helpers/reducer/utilsReducer.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
+const utilsReducer = (utilsState, { type }) => {
+
+    switch(type) {
+        case'HIDE_ASIDE':
+        return { ...utilsState, hideAside: !utilsState.hideAside}
+
+        // case'PIN_NOTE':
```
Comment (aurobindoGupta) [inline src/helpers/reducer/utilsReducer.js:7]: remove

File: src/index.js [hunk 12]
```diff
@@ -3,14 +3,17 @@ import ReactDOM from "react-dom";
 import App from "./App";
 import { makeServer } from "./server";
 import { BrowserRouter as Router } from "react-router-dom";
+import { NoteProvider } from "./helpers/context/note-context";
 
 // Call make Server
 makeServer();
```
Comment (aurobindoGupta) [inline src/index.js:9]: start using prettier

File: src/styles/homepage/create_notes.css [hunk 13]
```diff
@@ -3,8 +3,11 @@
     display: flex;
```
Comment (aurobindoGupta) [inline src/styles/homepage/create_notes.css:3]: if flex is used too many times make it a class

**Review by aurobindoGupta — APPROVED** [review]
nice work

**Pull request merged**